### PR TITLE
Add HelperOptions section

### PR DESCRIPTION
**Pull request by jf-viguier**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Add HelperOptions section to explain how to manage configuration form in modules.
| Fixed ticket? | none

### CODE REVIEW

File: src/content/1.7/modules/creation/adding-configuration-page.md [hunk 1]
```diff
@@ -320,3 +320,100 @@ in the back office:
 Change the value to whichever you like, click on the "Save" button, then
 go reload the homepage: your module is indeed updated with the new
 string!
+
+Using HelperOptions
+----------------
+
+THe `HelperOptions` helper is the recommanded one for managing configurations i.e. values stored in the configuration table.
```
Comment (Progi1984) [inline src/content/1.7/modules/creation/adding-configuration-page.md:327]: Where is it recommended ?

File: src/content/1.7/modules/creation/adding-configuration-page.md [hunk 2]
```diff
@@ -320,3 +320,100 @@ in the back office:
...
+<?php
+$this->fields_options = array(
+  'general' => array(
+    ['title'] => $this->l('Carrier options'),                      // The title of the fieldset. If missing, default is 'Options'.
+    ['top'] => $this->l('Text to display before the fieldset'),    // This text is display right above the first. Rarely used.
+    ['image'] => 'url to icon',                                    // If missing, will use the default icon for the tab.
+    ['description'] => $this->l('Display as description'),         // Displays an informational box above the fields.
+    ['info'] => $this->l('Display as info'),                       // Displays an unstyled text above the fields.
+    'fields' => array(                                             // The various option fields.
+      'PS_CARRIER_DEFAULT' => array(                               // The aray is named after the option's ID. It must be the
```
Comment (Progi1984) [inline src/content/1.7/modules/creation/adding-configuration-page.md:343]:
```suggestion
      'PS_CARRIER_DEFAULT' => array(                               // The array is named after the option's ID. It must be the
```

File: src/content/1.7/modules/creation/adding-configuration-page.md [hunk 3]
```diff
@@ -320,3 +320,100 @@ in the back office:
...
+    'title' => $this->module->l('My config title'),
+    'validation' => 'isInt',
+    'choices' => [
+        'myconfigvalue1' => $this->module->l('My value 1'),
+        'myconfigvalue2' => $this->module->l('My value 2'),
+    ],
+],
+```
+
+### checkbox example
```
Comment (Progi1984) [inline src/content/1.7/modules/creation/adding-configuration-page.md:391]:
```suggestion
### Checkbox example
```

File: src/content/1.7/modules/creation/adding-configuration-page.md [hunk 4]
```diff
@@ -320,3 +320,100 @@ in the back office:
...
+    'title' => $this->module->l('My config title'),
+    'validation' => 'isInt',
+    'choices' => [
+        'myconfigvalue1' => $this->module->l('My value 1'),
+        'myconfigvalue2' => $this->module->l('My value 2'),
+    ],
+],
+```
+
+### select example
```
Comment (Progi1984) [inline src/content/1.7/modules/creation/adding-configuration-page.md:405]:
```suggestion
### Select example
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: src/content/1.7/modules/creation/adding-configuration-page.md (hunk 1 above)
Comment (jf-viguier) [reply src/content/1.7/modules/creation/adding-configuration-page.md:327]: it's recommended by @Matt75

Comment (jf-viguier) [reply src/content/1.7/modules/creation/adding-configuration-page.md:327]: @Progi1984 you can see it there in action https://github.com/Matt75/displayordercarrier/blob/master/controllers/admin/AdminDisplayOrderCarrierController.php

**Comment by Progi1984** [comment]
@jf-viguier Could you check your modification in the PR #1074 (from @eternoendless) ?

**Comment by jf-viguier** [comment]
@Progi1984 my version of HelperForm is better : more complete, array notation, exemples.
If you merge  #1074 I can move this par here.

**Review by Sinepel — APPROVED** [review]
Thanks @jf-viguier !

File: src/content/1.7/modules/creation/adding-configuration-page.md (hunk 1 above)
Comment (kpodemski) [reply src/content/1.7/modules/creation/adding-configuration-page.md:327]: I wouldn't say that it is a recommended way as it's legacy part of the code, it is for legacy part, but for sure not for a modern architecture :) we need to inform developers about it

Comment (jf-viguier) [reply src/content/1.7/modules/creation/adding-configuration-page.md]: Thanks @kpodemski I've changed this.

**Comment by eternoendless** [comment]
Now that #1074 has been merged, would you kindly update your PR so that the content is on the right page? Thanks!

**Comment by matks** [comment]
Hello,

I have modified your PR to target the `1.7.x` branch.

We have modified the devdocs inner working in order to welcome multiple documentation versions. This is because soon we will have a 1.7 documentation and a 8.0 documentation, matching PrestaShop 1.7 or PrestaShop 8.0 softwares.

Read more on https://devdocs.prestashop.com/1.7/contribute/documentation/how/documentation-versions/

Please also note that now that we have multiple versions inside one documentation, we had to build a workflow that can handle this setup. This repository https://github.com/PrestaShop/docs now only contains the content of the documentation, and https://github.com/PrestaShop/devdocs-site contains the sources and the deployment workflow.

Because of this, some files have moved and you might have to rebase your Pull Request to match the new structure of the repository.

**Comment by PierreRambaud** [comment]
ping @jf-viguier  friendly reminder :heart:

**Comment by jf-viguier** [comment]
@PierreRambaud @matks Rebase stuff is broken, I've created a new pr here https://github.com/PrestaShop/docs/pull/1185

**Comment by Progi1984** [comment]
@jf-viguier I close this one.

**Pull request closed without merging**